src/Features/Core/Portable/CodeRefactorings/CodeRefactoringContextExtensions.cs

```diff
@@ -25,13 +30,33 @@ internal static class CodeRefactoringContextExtensions
             }
         }
 
-        internal static Task<TSyntaxNode> TryGetSelectedNodeAsync<TSyntaxNode>(this CodeRefactoringContext context)
+        internal static async Task<TSyntaxNode> TryGetSelectedNodeAsync<TSyntaxNode>(this CodeRefactoringContext context)
             where TSyntaxNode : SyntaxNode
         {
-            var document = context.Document;
-            var helpers = document.GetLanguageService<IRefactoringHelpersService>();
+            (var document, var span, var cancellationToken) = context;
+            var potentialNodes = await GetRelevantNodes<TSyntaxNode>(document, span, cancellationToken).ConfigureAwait(false);
+            return potentialNodes.FirstOrDefault();
+        }
+
+        internal static async Task<TSyntaxNode> TryGetSelectedNodeAsync<TSyntaxNode>(this Document document, TextSpan span, Func<TSyntaxNode, bool> predicate, CancellationToken cancellationToken)
+where TSyntaxNode : SyntaxNode
+        {
+            var potentialNodes = await GetRelevantNodes<TSyntaxNode>(document, span, cancellationToken).ConfigureAwait(false);
+            return potentialNodes.FirstOrDefault(predicate);
+        }
 
-            return helpers.TryGetSelectedNodeAsync<TSyntaxNode>(document, context.Span, context.CancellationToken);
+        internal static async Task<TSyntaxNode> TryGetSelectedNodeAsync<TSyntaxNode>(this Document document, TextSpan span, CancellationToken cancellationToken)
```

Comment:
can dleegate to method above with `a => true` predicate.

Reply:
Idk, it's shorter but IMO not more readable. If there were more overloads I'd agree, not sure like this.

Reply:
K.  Your call here.  My personal bar is: if i had to update one, would i need to update the other?  if so, just delegate.  But they're one liners, so i don't feel hugely about this.